src/middleware/jirahost-middleware.ts

```diff
@@ -19,10 +19,21 @@ const extractUnsafeJiraHost = (req: Request): string | undefined => {
 		return req.body?.jiraHost;
 	} else if (req.cookies.jiraHost) {
 		return req.cookies.jiraHost;
+	} else if (req.query?.jwt) {
+		return getJiraHostFromJwtToken(req.query.jwt as string);
```

Comment:
I assume this jiraHost comes from the asymmetric? I am afraid it may fail, as later on the same middleware verify it as symmetric?

Maybe we should fix this middleware and the new jwt middleware with asymmetric as well (and base on @mboudreau 's [PR](https://github.com/atlassian/github-for-jira/pull/1871)?